### ISSUE TITLE
`getUsage` does not work as documented

### ISSUE THREAD

**Issue by davelosert** [thenativeweb/command-line-interface#233]
## Problem

When using `getUsage` within a `handle`-Function, I get the following error:

```log
TypeError: Cannot read property 'map' of undefined
    at Object.getCommandsByPath (/Users/davidlosert/Projects/yabai-configurator/node_modules/command-line-interface/build/lib/getCommandsByPath.js:6:22)
    at Object.getCommandLineUsageConfiguration (/Users/davidlosert/Projects/yabai-configurator/node_modules/command-line-interface/build/lib/usage/getCommandLineUsageConfiguration.js:12:48)
```

this is my `handle`-Function:

```javascript
  async handle({ options, getUsage, ancestors }) {
    const subCommandName = await buntstift.select('Select Subcommand', Object.values(this.subcommands).map(command => command.name));

    if (subCommandName === 'help') {
      buntstift.info(getUsage([...ancestors, 'yco']));
      process.exit(0);
    }

    await runCli({ rootCommand: this.subcommands[subCommandName], argv: process.argv });
  }
```

## Cause
I investigated the issue and found the reason:
The `getUsage`-Function (defined within [getGetUsage](https://github.com/thenativeweb/command-line-interface/blob/master/lib/usage/getGetUsage.ts#L9)) actually expects an object with the property `commandPath` rather than an array (as documented):

```typescript
const getGetUsage = function ({ rootCommand }: {
  rootCommand: Command<any>;
}): GetUsageFn {
  return ({ commandPath }): string => {
    const commandLineUsageConfiguration = getCommandLineUsageConfiguration({ rootCommand, commandPath });

    return commandLineUsage(commandLineUsageConfiguration);
  };
};
```

As I am not sure which is supposed to be right (the documented way or the programmed), I did not send a PR yet.
But I am happy to once you tell me which I should fix. 😃

**Comment by goloroden** [comment]
Hey @davelosert 👋 

Sorry for the late answer, but better late than never 😉 

I've taken a look at how things work, and the programmed way is the correct one.
So, if you would like to come up with a PR, I'd love to merge it 😊 

Oh, and don't forget to add yourself as a contributor to the `package.json` file in your PR!

Have a nice day,


Golo

**Comment by davelosert** [comment]
@goloroden : No Problem, all is good. It was christmas after all 😅 .

I opened a PR: #236 .

**Comment by yeldiRium** [comment]
@davelosert after #236 is merged, is this issue resolved?

**Comment by goloroden** [comment]
@yeldiRium Since @davelosert discussed this with me, I can say that yes, this is resolved 😊 

So I'm going to close this (should have already done that, but apparently I had forgotten about it).
